class/defaults.yml

```diff
@@ -1,4 +1,10 @@
 parameters:
   openshift4_api:
     =_metadata: {}
-    namespace: syn-openshift4-api
+    additionalCORSAllowedOrigins: []
+    audit:
+      profile: 'Default'
+    clientCA: {}
+    encryption: {}
+    tlsSecurityProfile: {}
+    servingCerts: {}
```

Comment:
Looks like only `audit` and `servingCerts` are used out of the available parameters. This will lead to unwelcome surprises when users start configuring the other parameters and nothing happens.

To avoid this situation (and to future-proof our components) we generally prefer to provide a parameter which is merged into the resulting resource's `spec` instead of listing all the possible fields of the resource's `spec` as individual parameters.

With this approach, you can still expose `servingCerts` as a separate parameter which is handled with rich semantics and merged over the contents for `spec` which were provided by the user.

Reply:
That the other parameters have not been used is an oversight on my part and definitely not intentional... I seem to have been to focused on `servingCerts`

If I understood you correctly, you would just introduce a `spec` parameter that will be merged into the apiserver config?

Reply:
I'd probably pick `apiServerSpec` for the parameter name, and have roughly the following:

```
parameters:
  openshift4_api:
    servingCerts: {} # rendered with the same logic as previously
    apiServerSpec:
      audit:
        profile: Default
```

and then in the code

```
apiserver = {
  ...
  metadata: { ... }
  spec: com.makeMergeable(params.apiServerSpec) + {
    servingCerts: [
      # current logic
    ]
  }
}